skll/data/readers.py

```diff
@@ -878,67 +939,72 @@ def split_with_quotes(self,
-        # additional arguments we want
-        # to pass to the `pd.read_csv()` function
-        kwargs = {'header': header,
-                  'delimiter': delimiter,
-                  'quotechar': quote_char,
-                  'escapechar': escape_char}
-        kwargs.update(self._pandas_kwargs)
-
-        df = pd.read_csv(StringIO(s), engine=self._engine, **kwargs)
-        return df
+        return next(csv.reader([s], delimiter=delimiter, quotechar=quote_char,
+                               escapechar=escape_char))
 
-    def _sub_read(self, path):
+    def _sub_read(self, f):
         """
         Parameters
         ----------
-        path : str
-            The path to the ARFF file.
+        f : file buffer
+            A file buffer for the ARFF file.
 
-        Returns
-        -------
-        ids : np.array
-            The ids for the feature set.
-        labels : np.array
-            The labels for the feature set.
-        features : list of dicts
-            The features for the features set.
+        Yields
+        ------
+        curr_id : str
+            The current ID for the example.
+        class_name : float or str
+            The name of the class label for the example.
+        example : dict
+            The example valued in dictionary format, with 'x'
+            as list of features.
         """
-        with open(path, 'r') as buff:
-            lines = [UnicodeDammit(line.strip(), ['utf-8', 'windows-1252']).unicode_markup
-                     if not isinstance(line, str)
-                     else line.strip() for line in buff if line.strip()]
-
-        # find the row index starting with data; the line below this
-        # is where the data should actually begin
-        data_idx = lines.index('@data')
-
-        df = self.split_with_quotes('\n'.join(lines[data_idx + 1:]), delimiter=',')
-
-        # get the column names from the attribute
-        # rows, and add them to the columns list
-        columns = []
-        for row in lines[:data_idx]:
-
-            row_series = self.split_with_quotes(row)
-            if row_series.loc[0, 0] == '@attribute':
-                column = row_series.loc[0, 1]
-                columns.append(column)
-
-                # if the column is the label column,
-                # and the type is 'numeric', set regression
-                # to True; otherwise False
-                if column == self.label_col:
-                    self.regression = row_series.loc[0, 2] == 'numeric'
-
-            # if the relation attribute exists, then
-            # add it to the relation instance variable
-            elif row_series.loc[0, 0] == '@relation':
-                self.relation = row_series.loc[0, 1]
-
-        df.columns = columns
-        return self._parse_dataframe(df, self.id_col, self.label_col)
+        field_names = []
+        # Process ARFF header
+        for line in f:
+            # Process encoding
+            if not isinstance(line, text_type):
```

Comment:
I thought all these were already removed before, but seems there are still leftovers.

Reply:
These came back because @jbiggsets is trying to revert the changes he made earlier and he just happened to use a branch where your 2.7-removal changes had not been merged in.